streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -97,4 +97,11 @@
      */
     KeyValueIterator<K, V> all();
 
+    /**
+     * Return the number of key-value mappings in this store.
+     *
+     * @return the number of key/value pairs in the store.
+     */
+    int size();
```

Comment:
`approximate` sounds good. Also to differentiate with number of bytes as @jkreps  pointed out on the ticket, how about use `approximateNumEntries`?